runelite-client/src/main/java/net/runelite/client/plugins/nightmarezone/NightmareZonePlugin.java

```diff
@@ -0,0 +1,161 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.nightmarezone;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Binder;
+import com.google.inject.Provides;
+import net.runelite.api.*;
+import net.runelite.api.events.ChatMessage;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.Notifier;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.ClientUI;
+import net.runelite.client.ui.overlay.Overlay;
+
+import javax.inject.Inject;
+
+@PluginDescriptor(
+		name = "Nightmare Zone Plugin"
+)
+public class NightmareZonePlugin extends Plugin
+{
+	@Inject
+	private Notifier notifier;
+
+	@Inject
+	private ClientUI gui;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private NightmareZoneConfig config;
+
+	@Inject
+	private NightmareZoneOverlay overlay;
+
+	@Override
+	public void configure(Binder binder)
+	{
+		binder.bind(NightmareZoneOverlay.class);
+	}
+
+	@Override
+	public Overlay getOverlay()
+	{
+		return overlay;
+	}
+
+	// This starts as true since you need to get
+	// above the threshold before sending notifications
+	private boolean absorptionNotificationSend = true;
+
+	@Provides
+	NightmareZoneConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(NightmareZoneConfig.class);
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (!config.enabled())
+		{
+			return;
+		}
+		if (config.absorptionNotification())
+		{
+			checkAbsorption();
+		}
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (!config.enabled())
+		{
+			return;
+		}
+
+		String msg = event.getMessage().replaceAll("<[^>]*>", " "); //remove color and linebreaks
+		if (config.overloadNotification() && msg.contains("The effects of overload have worn off, and you feel normal again."))
```

Comment:
Should this check that you are actually within nmz? Also you must check message type or else players ingame can trigger it via public chat.